test/integration_test/fafb_matrix_test.go

```diff
@@ -81,21 +88,32 @@ var flashArrayPositiveInstallTests = []PureTestrailCase{
 	// C55003: https://portworx.testrail.net/index.php?/cases/view/55003
 	// Uninstall PX through operator with Pure arrays, after clean install
 	{
-		TestCase: TestCase{
-			TestrailCaseIDs:         []string{"C55129", "C56414", "C56411", "C55003"},
-			Spec:                    simpleClusterSpec,
-			ShouldStartSuccessfully: true,
+
+		TestName:        "C55129-C56414-C56411-C55003",
+		TestrailCaseIDs: []string{"C55129", "C56414", "C56411", "C55003"},
+		TestSpec: func(t *testing.T) interface{} {
+			// Construct Portworx StorageCluster object
+			cluster := &v1.StorageCluster{}
+			cluster.Spec = simpleClusterSpec
+			err := constructStorageCluster(cluster, pxSpecGenURL, pxSpecImages)
```

Comment:
You can define this whole thing as one function that will return a function and then reuse it between all the test cases, like this:

```
func createPureTestSpecFunction(cluster *v1.StorageCluster) func(t *testing.T) interface{} {
    return func(t *testing.T) {
        // The same function block as before
    }
}
```
and then invoke as
```
TestSpec: createPureTestSpecFunction(&v1.StorageCluster{
  Spec: simpleClusterSpec
})
```

This will help reduce code duplication and make it a lot more readable